lib/livebook_web/live/hub/edit_live.ex

```diff
@@ -98,29 +89,25 @@ defmodule LivebookWeb.Hub.EditLive do
   def handle_info({:secret_created, %{hub_id: id}}, %{assigns: %{hub: %{id: id}}} = socket) do
     {:noreply,
      socket
-     |> refresh_secrets()
+     |> push_redirect(to: ~p"/hub/#{id}")
```

Comment:
I think you can make this work by having an additional `counter`. Every time you want to refresh, you bump the counter. You will also pass the counter in the `live_component` call. The component itself doesn't need to use the counter. This way you don't need to redirect.

If that doesn't work though, the redirect is fine. :)